Hymns/Favorites/FavoriteEntity.swift

```diff
@@ -0,0 +1,19 @@
+import SwiftUI
+import RealmSwift
+
+class FavoriteEntity: TagEntity {
+
+    private static let favoriteTag = "_*_favorited_*_"
+
+    required init() {
+        super.init()
+    }
+
+    init(hymnIdentifier: HymnIdentifier, songTitle: String) {
+        super.init(hymnIdentifier: hymnIdentifier, songTitle: songTitle, tag: Self.favoriteTag)
+    }
+
+    static func createPrimaryKey(hymnIdentifier: HymnIdentifier) -> String {
+        return TagEntity.createPrimaryKey(hymnIdentifier: hymnIdentifier, tag: favoriteTag)
+    }
```

Comment:
Maybe we can revert this file too, instead of piggyback off of TagEntity. It seems like this current behavior is just lucky and isn't technically supported by Realm, so i'm scared that they'll change this undefined behavior somewhere down the line and our stuff will break.

Reply:
Hahaha lol! We have gone full circle.... xD   Now there is basically no need of even having a favoriteTag. We might as well go back to how we had it where just delete objects when they are unfavorited and not even store the variable favorite tag? Is that what you had in mind too?

Reply:
Addressed in recent commit